cot/src/db/query.rs

```diff
@@ -137,6 +191,25 @@ impl<T: Model> Query<T> {
         db.get(self).await
     }
 
+    /// Execute the query and return the number of results.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the query fails.
+    pub async fn count(&self, db: &Database) -> db::Result<usize> {
+        let mut select = sea_query::Query::select();
+        select
+            .from(T::TABLE_NAME)
+            .expr(sea_query::Expr::cust("COUNT(*)"));
```

Comment:
We prefer not to use raw SQL if avoidable using sea_query to make this as portable between SQL engines as possible. This could be expressed as so:

```suggestion
            .expr(sea_query::Expr::col(sea_query::Asterisk).count());
```